File: competitions/20200501/I.cpp
/************************************************
 * TODO Auther: by SR
 * FilePath: \digital designc:\Users\SR\Documents\vscodefile\ACM\competitions\20200501\I.cpp
 * Date: 2020-05-01 12:50:58
 * LastEditTime: 2020-05-01 12:55:11
 * !今日运势: 吉，无bug
************************************************/
#include<iostream>
#include<cmath>
#include<algorithm>
#include<cstdio>
#include<string.h>
using namespace std;
typedef long long ll;

int main(){

    int n,que[20],length;
    cin>>n;
    for(int i=0; i<n; i++)
        cin>>que[i];
	cin>>length;
    
    int ans,MIN=0x3f3f3f3f;
    for(int i=0; i<n; i++){
        (length%que[i]<MIN)?(ans=que[i], MIN=length%que[i]):1;
    }
    cout<<ans;

	//cout<<"\n";
	//system("pause");
	return 0;
}

File: competitions/20200607/I.py
'''
@TODO Auther: by SR
@FilePath: \digital designc:\Users\SR\Documents\vscodefile\ACM\competitions\20200607\I.py
@Date: 2020-06-07 16:15:59
@LastEditTime: 2020-06-08 11:08:31
@!今日运势: 吉，无bug
'''
(n, k) = map(int, input().split())
arr = []
def takeSecond(elem):
    return elem[1]
for i in range(n):
    (name, num) = input().split()
    arr.append((name, int(num)))
arr.sort(key = takeSecond, reverse = True)
ans = []
for i in range(n):
    if k >= arr[i][1]:
        k = k - arr[i][1]
        ans.append(arr[i][0])
if k == 0:
    print(len(ans))
    for i in range(len(ans)):
        print(ans[i])
else:
    print(0)